otus-06/src/otus_06/homework.clj

```diff
@@ -25,36 +86,152 @@
+    (println line)))
+
 ;; 3. Display Sales Table
+(defn display-sales
+  []
+  (let [customers (into {}
+                        (for [customer (get-customers)
+                              :let [{:keys [custID name]} customer]]
+                          [custID name]))
+        products (into {}
+                       (for [product (get-products)
+                             :let [{:keys [prodID itemDescription]} product]]
+                         [prodID itemDescription]))]
+    (doseq [line (for [sales (get-sales)
+                       :let [{:keys [custID prodID itemCount]} sales
+                             client (get customers custID)
+                             product (get products prodID)]]
+                   (format "Client: %s, Product: %s, Count: %s"
+                           client
+                           product
+                           itemCount))]
+      (println line))))
+
 ;; 4. Total Sales for Customer
+(defn show-sales-customer
+  "Supports showing result for multiple customers."
+  []
+  (print "Enter customer name: ")
+  (flush)
+  (let [name (my-read-line)
+        product-prices (into {}
+                             (for [product (get-products)
+                                   :let [{:keys [prodID unitCost]} product]]
+                               [prodID unitCost]))
+        sales (apply merge-with
+                +
+                (for [customer (get-customers :name name)
+                      :let [{:keys [custID name]} customer]
+                      sales (get-sales :custID custID)
+                      :let [{:keys [itemCount prodID]} sales
+                            sum (* (Double. itemCount)
+                                   (Double. (get product-prices prodID)))]]
+                  {name sum}))
+        sales-with-default (if (seq sales) sales {"Not found" 0})]
+    (doseq [line sales-with-default]
+      (println (format "Name: %s, Sum: $%s" (key line) (val line))))))
+
 ;; 5. Total Count for Product
+
+(defn show-sales-product
+  []
+  (print "Enter product name: ")
+  (flush)
+  (let [item-desc (my-read-line)
+        sales (apply merge-with
+                +
+                (for [product (get-products :itemDescription item-desc)
+                      :let [{:keys [prodID itemDescription]} product]
+                      sales (get-sales :prodID prodID)
+                      :let [{:keys [itemCount]} sales]]
+                  {itemDescription (Double. itemCount)}))
+        sales-with-default (if (seq sales) sales {"Not found" 0})]
+    (doseq [line sales-with-default]
+      (println (format "Product: %s, Count sold: %s" (key line) (val line))))))
 ;; 6. Exit
+(defn exit [] (println "Bye!") (System/exit 0))
 
 ;; Enter an option?
 
+(defn menu
+  []
+  (clear-screen)
+  (doseq [menu ["1. Display Customer Table" "2. Display Product Table"
+                "3. Display Sales Table" "4. Total Sales for Customer"
+                "5. Total Count for Product" "6. Exit"]]
```

Comment:
Dude, make it readable, split on different lines :)
```
"1. ...
"2. ...
...
"

Reply:
I don't know. But auto formatter did that